File: 2016/fase1/chuva/chuva.js
function secoesCobertasPorAgua(N, alturas) {
    let secoesCobertas = 0;
    let maxAltura = 0;

    // Encontrar a altura máxima
    for (let i = 0; i < N; i++) {
        maxAltura = Math.max(maxAltura, alturas[i]);
    }

    // Contar seções cobertas por água
    for (let i = 0; i < N; i++) {
        if (alturas[i] >= maxAltura) {
            secoesCobertas++;
        }
    }

    return secoesCobertas;
}

// Entrada de exemplo
const N = 6;
const alturas = [2, 5, 3, 2, 6, 3];

// Chamada da função e exibição do resultado
const resultado = secoesCobertasPorAgua(N, alturas);
console.log(resultado); // Saída: 2

File: 2016/fase1/chuva/chuva.py
def secoes_cobertas_por_agua(N, alturas):
    cobertas_por_agua = 0
    max_altura = 0
    
    for altura in alturas:
        if altura > max_altura:
            max_altura = altura
            cobertas_por_agua += 1
    
    return cobertas_por_agua

# Leitura da entrada
N = int(input())
alturas = [int(input()) for _ in range(N)]

# Chamada da função e impressão do resultado
print(secoes_cobertas_por_agua(N, alturas))